pkg/operator/cluster/ceph_user.go

```diff
@@ -0,0 +1,68 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/ceph/client"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func newCephUser(context *clusterd.Context, ns string) *cephUser {
+	n := fmt.Sprintf("%s-rook-user", ns)
+	u := "client." + n
+	return &cephUser{
+		secretName:       n,
+		username:         u,
+		clusterNamespace: ns,
+		context:          context,
+		access:           []string{"osd", "allow rwx", "mon", "allow r"},
+	}
+}
+
+type cephUser struct {
+	secretName       string
+	username         string
+	access           []string
+	context          *clusterd.Context
+	clusterNamespace string
+	key              string
+}
+
+func (cu *cephUser) create() error {
+	key, err := client.AuthGetOrCreateKey(cu.context, cu.clusterNamespace, cu.username, cu.access)
+	if err != nil {
+		return fmt.Errorf("failed to get or create auth key for %s. %+v", cu.username, err)
+	}
+
+	cu.key = key
```

Comment:
That is an attempt to simplify it for the caller, can't think of a use case where caller would not want to store it. How about doing both, returning the key and storing it to give the option to the caller to handle it manually?